service/history/historyEngine.go

```diff
@@ -1021,6 +1026,32 @@ func (c *decisionBlobSizeChecker) failWorkflowIfBlobSizeExceedsLimit(blob []byte
 	return true, nil
 }
 
+func (c *decisionBlobSizeChecker) failWorkflowIfTotalSizeExceedsLimit() error {
+	err := common.CheckEventBlobSizeLimit(
```

Comment:
I'm not convinced enforcing this limit by accumulating the user blob sizes is the right approach.  This is more like system protection and needs to be enforced as the total size of transaction when we are appending history events to events table.